scripts/ioc_deploy.py

```diff
@@ -0,0 +1,327 @@
+            return ReturnCode.NO_CONFIRM
+    if Path(deploy_dir).exists():
+        logger.info(f"{deploy_dir} exists, skip git clone step.")
+    else:
+        rval = clone_repo_tag(
+            name=upd_name,
+            github_org=args.github_org,
+            release=upd_rel,
+            deploy_dir=deploy_dir,
+            dry_run=args.dry_run,
+        )
+        if rval != ReturnCode.SUCCESS:
+            logger.error(f"Nonzero return value {rval} from git clone")
+            return rval
+
+    logger.info(f"Building IOC at {deploy_dir}")
+    rval = make_in(deploy_dir=deploy_dir, dry_run=args.dry_run)
+    if rval != ReturnCode.SUCCESS:
+        logger.error(f"Nonzero return value {rval} from make")
+        return rval
+    return ReturnCode.SUCCESS
+
+
+def finalize_name(name: str, github_org: str) -> str:
+    """
+    Check if name is present in org and is well-formed.
+
+    If the name is present, return it.
+    If the name is not present and the correct name can be guessed, guess.
+    If the name is not present and cannot be guessed, raise.
+
+    A name is well-formed if it starts with "ioc", is hyphen-delimited,
+    and has at least three sections.
+
+    For example, "ioc-common-gigECam" is a well-formed name for the purposes
+    of an IOC deployment. "ads-ioc" and "pcdsdevices" are not.
+
+    However, "ads-ioc" will resolve to "ioc-common-ads-ioc".
+    Only common IOCs will be automatically discovered using this method.
+    """
+    split_name = name.split("-")
+    if len(split_name) < 3 or split_name[0] != "ioc":
+        new_name = f"ioc-common-{name}"
+        logger.warning(f"{name} is not an ioc name, trying {new_name}")
+        name = new_name
+    logger.debug(f"Checking for {name} in org {github_org}")
+    try:
+        resp = urllib.request.urlopen(f"https://github.com/{github_org}/{name}")
+        if resp.code == 200:
+            logger.info(f"{name} exists in {github_org}")
+            return name
+        else:
+            logger.error(f"Unexpected http error code {resp.code}")
+    except urllib.error.HTTPError as exc:
+        logger.error(exc)
+        logger.debug("", exc_info=True)
+    raise ValueError(f"{name} does not exist in {github_org}")
+
+
+def finalize_tag(name: str, github_org: str, release: str) -> str:
+    """
+    Check if release is present in the org.
+
+    We'll try a few prefix options in case the user has a typo.
+    In order of priority with no repeats:
+    - user's input
+    - R1.0.0
+    - v1.0.0
+    - 1.0.0
+    """
+    try_release = [release]
+    if release.startswith("R"):
+        try_release.extend([f"v{release[1:]}", f"{release[1:]}"])
+    elif release.startswith("v"):
+        try_release.extend([f"R{release[1:]}", f"{release[1:]}"])
+    elif release[0].isalpha():
+        try_release.extend([f"R{release[1:]}", f"v{release[1:]}", f"{release[1:]}"])
+    else:
+        try_release.extend([f"R{release}", f"v{release}"])
```

Comment:
I feel like this could be more ~~concise~~ unrelentingly pythonic,  maybe something like:
```python
try_release = [f"{pre}{release[1:]" for pre in ["R", "v", ""]]
bubble_index = next(x for x in range(try_release) if try_release[x].startswith(release[0]), None)

if bubble_index is None:
    # handle alpha, or do nothing
else:
    try_release.insert(0, try_release.pop(bubble_index))
```

After ideating this I'm not sure it's better.  it's definitely not clearer haha

Reply:
Yeah I definitely had problems braining this properly to handle any or no prefix in a clean way. Probably there's a simple regex that would extract the three numbers and then try all the normal prefixes. I'm open to suggestions on other ways to implement it.